app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -1286,6 +1293,19 @@ class PluginApi {
     return modifySelectKit(pluginApiKey);
   }
 
+  /**
+   * Add a custom CSS class to the topic title within the header.
+   *
+   * ```javascript
+   * api.addHeaderFancyTitleClass('foo');
+   * api.addHeaderFancyTitleClass('bar');
+   * ```
+   *
+   **/
+  addHeaderFancyTitleClass(titleClass) {
```

Comment:
How do you imagine this being called? From inside some plugin outlet?

I feel like this would be better as a callback? So you could do something like this:

```js
addHeaderTopicTitleClass(
  (topic) => {
    if(topic.hasSomeValue){
      return ["my-class"];
    }
  }
)
```

Reply:
This is not needed, it is not being used anywhere